aten/src/ATen/core/ivalue_inl.h

```diff
@@ -345,7 +339,8 @@ struct C10_EXPORT ivalue::Future final : c10::intrusive_ptr_target {
    */
   c10::intrusive_ptr<Future> then(
       std::function<IValue(void)> callback,
-      TypePtr type) {
+      TypePtr type,
+      bool propagateTLSState = false) {
```

Comment:
Same as the comment in `addCallback`, would prefer not having this flag at all and make propagation the default.